src/CabanaPD_Solver.hpp

```diff
@@ -258,16 +271,23 @@ class SolverElastic
             // Integrate - velocity Verlet first half.
             integrator->initialHalfStep( *particles );
 
-            // Add non-force boundary condition.
-            if ( !boundary_condition.forceUpdate() )
-                boundary_condition.apply( exec_space(), *particles, step * dt );
+            // Update ghost particles.
+            comm->gatherDisplacement();
 
-            if constexpr ( std::is_same<typename force_model_type::thermal_type,
-                                        TemperatureDependent>::value )
+            if constexpr ( is_heat_transfer<
+                               typename force_model_type::thermal_type>::value )
+            {
+                if ( step % thermal_subcycle_steps == 0 )
+                    computeHeatTransfer( *heat_transfer, *particles, *neighbors,
```

Comment:
I am thinking that, if we use temperature b.c., that would need to be applied before solving the heat transfer, and maybe we will need to gather temperature before that as well. Thoughts?

Reply:
With force BC we purposely update forces, then apply BC so that it overrides the local update - don't we want the same thing here?

Reply:
This would require discussion. 

Here are two thoughts:
- For body forces, we first compute internal forces to then add the external body force. 
- For velocity b.c., in the KW example, we wanted to keep the velocity constant, so we overwrote the forces to make them 0, so that velocity won't change. 

These are different scenarios than my comment.

Reply:
BTW: is the temperature updated as part of the "non-force boundary condition"?

We may need the corresponding lines before the above addition as well. In this example, it may not be critical, because the temperature b.c. is constant.